[FIX] web_editor,website_sale: fix click on pricelist while editing

tlibioulle opened the pull request:
Prior to this commit, when a user edits the Shop page, clicks on the
pricelist dropdown and finally clicks elsewhere, a traceback occurs:
`Uncaught TypeError: Cannot read properties of undefined (reading
'getColorNames')`

This commit ensures the link tools is not toggles on dropdown links,
i.e. the customizable links selector does not contain the dropdown
anymore.

task-2695091

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

robodoo commented:
[Pull request status dashboard](https://mergebot.odoo.com/odoo/odoo/pull/84834).

tlibioulle commented:
Hi @bvr-odoo  @qsm-odoo ,

What do you think about it ? I fear the fix is a little bit too restrictive for other dropdown usages and I'm not comfortable enough with the editor to deal with maybe other options to add on the pricelist dropdown element.

Let me now if there are things I have to learn and apply to fix it.

Thanks in advance,

bvr-odoo commented:
> Hi @bvr-odoo @qsm-odoo ,
> 
> What do you think about it ? I fear the fix is a little bit too restrictive for other dropdown usages and I'm not comfortable enough with the editor to deal with maybe other options to add on the pricelist dropdown element.
> 
> Let me now if there are things I have to learn and apply to fix it.
> 
> Thanks in advance,

LGTM, because even if we leave the options on the dropdown links. It would first be necessary to fix all the related bugs (I tested 1 minute and there are several problems). cc @Zinston

tlibioulle commented:
> However, I tested your branch and there is still the popover that appears when you click on a dropdown link =>

@bvr-odoo Indeed, I just fixed the traceback but I noticed it would be better to exclude the dropdown item from the customizableLinksSelector aswell. I can add it if you think it makes sense.

Thanks a lot !

Zinston commented:
Do you mean you exclude a dropdown but not its dropdown items and they're all <a>s ? If so yes it would make sense to exclude the items too.
So I'm not sure I'm the person to ask because I don't know all the use cases in website, blog etc. but it makes sense to me to exclude these. Now what I'm wondering though is: are those dropdowns supposed to be editable at all? As in: can you click on them and change their text? Because if not, a more logical thing would be to exclude links that are not editable (`a.isContentEditable === false`).

tlibioulle commented:
@Zinston @bvr-odoo 

It seems the logic didn't exclude the contenteditable="false" yet, so I added it, thanks for this good input.

So in the last fixup I added : `.dropdown-item` and `contenteditable="false"` as selectors to exclude. 
Do you think this list of html attributes/class to exclude from the `customizableLinksSelector` should be more flexible ? Or do we continue to add those attributes like I do for the moment ?

Zinston commented:
I didn't mean `contenteditable="false"` but `isContentEditable === false` : https://developer.mozilla.org/en-US/docs/Web/API/HTMLElement/isContentEditable

Williambraecky reviewed (approved):
LGTM

Williambraecky commented:
@tlibioulle This needs a rebase + squash :)

tivisse commented:
@robodoo r+
